tests/cpp/unit-tests/test_addressatcommand.cpp

```diff
@@ -0,0 +1,108 @@
+#include "../../../third-party/catch/catch.hpp"
+#include "addressatcommand.h"
+
+using namespace SmartRedis;
+
+SCENARIO("Ensuring the iterators for an AddressAtCommand are correct", "[AddressAtCommand]")
+{
+    GIVEN("An AddressAtCommand with a single field")
+    {
+        AddressAtCommand cmd;
+        std::string field = "INFO";
+        cmd.add_field(field);
+        WHEN("Iterators from the AddressAtKeyCommand object's "
+             "_fields data member are retrieved")
+        {
+            Command::iterator begin_it = cmd.begin();
+            Command::iterator end_it = cmd.end();
+            Command::const_iterator cbegin_it = cmd.cbegin();
+            Command::const_iterator cend_it = cmd.cend();
+            THEN("The iterators that were retrieved are the correct iterators")
+            {
+                CHECK(*cmd.begin() == field);
+                CHECK(*cmd.cbegin() == field);
+                begin_it++;
+                cbegin_it++;
+                CHECK(begin_it == end_it);
+                CHECK(cbegin_it == cend_it);
+            }
+        }
+    }
+}
+
+SCENARIO("Testing assignment operator for AddressAtCommand on heap", "[AddressAtCommand]")
+{
+    GIVEN("An AddressAtCommand object on the heap")
+    {
+        AddressAtCommand* cmd = new AddressAtCommand;
+
+        WHEN("Fields are added to the AddressAtCommand in every possible manner")
+        {
+            // define fields
+            std::string field_1 = "INFO";
+            char field_2[8] = "CLUSTER";
+            const char field_3[11] = "everything";
+            char field_4[5] = "INFO";
+            size_t field_size_4 = std::strlen(field_4);
+            std::string field_5 = "INFO";
+            std::vector<std::string> fields_1 = {"CLUSTER", "everything", "INFO"};
+            std::string_view field_sv = std::string_view(field_4, field_size_4);
+
+            // define necessary variables for testing
+            std::string output = " INFO CLUSTER everything INFO "
```

Comment:
Is the ``Command`` ``to_string()`` prepending with a space and that's why we need to format like this?